### PR TITLE
Async socket unittest and specs

### PR DESCRIPTION
**Pull request by jebrando**

### CODE REVIEW

File: linux/devdoc/async_socket_linux.md [hunk 1]
```diff
@@ -0,0 +1,336 @@
+﻿# `async_socket_linux` requirements
+
+## Overview
+
+`async_socket_win32` is an implementation of `async_socket` that wraps the asynchronous socket APIs for Windows.
```
Comment (mattdurak) [inline linux/devdoc/async_socket_linux.md:5]: >Windows [](http://example.com/codeflow?start=104&length=7)

Linux

File: linux/devdoc/file_linux_requirements.md [hunk 2]
```diff
@@ -78,65 +79,65 @@ MOCKABLE_FUNCTION(, void, file_destroy, FILE_HANDLE, handle);
...
+**NON_FILE_LINUX_43_031: [** If `handle` is `NULL` then `file_write_async` shall fail and return `FILE_WRITE_ASYNC_INVALID_ARGS`. **]**
 
-**SRS_FILE_LINUX_43_032: [** If `source` is `NULL` then `file_write_async` shall fail and return `FILE_WRITE_ASYNC_INVALID_ARGS`. **]**
+**NON_FILE_LINUX_43_032: [** If `source` is `NULL` then `file_write_async` shall fail and return `FILE_WRITE_ASYNC_INVALID_ARGS`. **]**
 
-**SRS_FILE_LINUX_43_051: [** If `position` + `size` is greater than `INT64_MAX`, then `file_write_async` shall fail and return `FILE_WRITE_ASYNC_INVALID_ARGS`. **]**
+**NON_FILE_LINUX_43_051: [** If `position` + `size` is greater than `INT64_MAX`, then `file_write_async` shall fail and return `FILE_WRITE_ASYNC_INVALID_ARGS`. **]**
 
-**SRS_FILE_LINUX_43_049: [** **SRS_FILE_LINUX_43_033: [** If `user_callback` is `NULL` then `file_write_async` shall fail and return `FILE_WRITE_ASYNC_INVALID_ARGS`. **]** **]**
+**NON_FILE_LINUX_43_049: [** **NON_FILE_LINUX_43_033: [** If `user_callback` is `NULL` then `file_write_async` shall fail and return `FILE_WRITE_ASYNC_INVALID_ARGS`. **]** **]**
```
Comment (darobs) [inline linux/devdoc/file_linux_requirements.md:88]: >NON_FILE_LINUX_43_033 [](http://example.com/codeflow?start=31&length=21)

This is twice the requirement of all the others!

Comment (jebrando) [reply linux/devdoc/file_linux_requirements.md:88]: :smile:

File: interfaces/devdoc/platform_requirements.md [hunk 3]
```diff
@@ -12,28 +11,28 @@ implementers
...
 The `platform_init` call performs any global initialization necessary for a particular platform.
 
 ```c
 int platform_init();
 ```
 
-**SRS_PLATFORM_30_000: [** The `platform_init` call shall perform any global initialization needed by the platform and return 0 on success. **]**
-
-**SRS_PLATFORM_30_001: [** On failure, `platform_init` shall return a non-zero value. **]**
+**NON_PLATFORM_30_000: [** The `platform_init` call shall perform any global initialization needed by the platform and return 0 on success. **]**
```
Comment (darobs) [inline interfaces/devdoc/platform_requirements.md:26]: >NON [](http://example.com/codeflow?start=2&length=3)

Why did all the "SRS get changed to "NON" in this PR?

Comment (jebrando) [reply interfaces/devdoc/platform_requirements.md:26]: These SRS don't correspond to code, but are here just in general and the traceability was flagging all these as missing in code and tests.  I didn't want to remove the files just yet until all the linux specs get written.

**Review by darobs — APPROVED** [review]
:shipit:

**Review by darobs — APPROVED** [review]

**Pull request merged**